README.md

```diff
@@ -60,7 +60,83 @@ Once you have implemented a lower bound search for the start of the range, imple
+
+      def lower_bound(x, v):
+         def search(arr, v, first, last):
+            if v > arr[last]: # 1
+                  return len(arr)
+
+            if last >= first: # 2
+                  mid = (last + first) // 2
+                  
+                  if mid == 0: # 3
+                     if arr[mid] >= v:
+                        return mid
+                     else:
+                        return mid + 1      
+
+                  if arr[mid] >= v: # 4
+                     # 5
+                     if arr[mid - 1] < v: # 6
+                        return mid
+                     else: # 7
+                        return search(arr, v, first, mid - 1) 
+                  else: # 8
+                     return search(arr, v, mid + 1, last)
+
+         first = 0
+         last = len(x) - 1
+         return search(x, v, first, last)
+
+We will go through the code one by one. I put the number so it can be easier to read.
+
+1. Base case: if the largest number in the array is smaller than integer we are looking for.
+2. We will run the code below if the last index is equal or larger than the first index. 
+3. We end up in this condition if we already reach the very left of array, where last = first = 0. If the value in that index is equal or larger than the number we are looking for, we return 0, otherwise we return 1.
+4. Otherwise, we do the binary search. We check if value in the mid index, is the same or larger than the number we search.
+5. If yes, then we check if the value in the index before the mid is smaller than the number we search. `This part is where it differs from the normal binary search.`
+6. If it also yes, this mid index is indeed the lower bound of the number we are looking for.
+7. Otherwise, we will continue doing a binary search in the  left part of mid index
+8. If the value in mid index we found is smaller than what we are looking for, so it must be in the right side of our mid index.
+
 *Would anything be more difficult if the features covered ranges instead of single nucleotides (like real BED files)? What could go wrong, if anything?*
+We are looking at the chrom_start for the features. This would ot change if the features wasn't SNPs. 
+All features with start within a certain region was extracted, and this would still be possible even if the features wasn't SNPs. 
+If we want the entire feature to be within the range given by the query file we would have to change the extract_features to include the end-position. 
+
+If the features wasn't SNPs we would also have to account for how we wanted them sorted if several features has the same start but different lengths. 
+The obvius would be to sort them with the shortest/them with the lowest chrom_end first. But we would have to include this in the code. 
+And the same thing for merging. 
```

Comment:
Correct.

Another issue is that an overlapping interval might not start or end anywhere near the query interval. Long intervals could start long before and end long after.